rasterio/coords.py

```diff
@@ -43,5 +43,16 @@ def disjoint_bounds(bounds1, bounds2):
     ``True`` if bounds are disjoint,
     ``False`` if bounds overlap
     """
-    return (bounds1[0] > bounds2[2] or bounds1[2] < bounds2[0] or
-            bounds1[1] > bounds2[3] or bounds1[3] < bounds2[1])
+    bounds1_north_up = bounds1[3] > bounds1[1]
+    bounds2_north_up = bounds2[3] > bounds2[1]
+
+    if not bounds1_north_up and bounds2_north_up:
+        # or both south-up (also True)
+        raise ValueError("Bounds must both have the same orientation")
+
+    if bounds1_north_up:
+        return (bounds1[0] > bounds2[2] or bounds2[0] > bounds1[2] or
+                bounds1[1] > bounds2[3] or bounds2[1] > bounds2[3])
+    else:
+        return (bounds1[0] > bounds2[2] or bounds2[0] > bounds1[2] or
+                bounds1[3] > bounds2[1] or bounds2[3] > bounds2[1])
```

Comment:
note: I previously ran our CLI tools on an north-down image and made a note of the results, we should update this table using the new code

 https://github.com/mapbox/rasterio/issues/525#issuecomment-206368355